cmd/completion.go

```diff
@@ -0,0 +1,267 @@
+package cmd
+
+import (
+	"bytes"
+	"fmt"
+	"io"
+	"os"
+	"strings"
+
+	"github.com/pkg/errors"
+	"github.com/spf13/cobra"
+)
+
+
+const completionDesc = `
+Generate autocompletions script for Astro for the specified shell (bash or zsh).
+
+This command can generate shell autocompletions. e.g.
+
+	$ astro completion bash
+
+Can be sourced as such
+
+	$ source <(astro completion bash)
+`
+
+var (
+	completionShells = map[string]func(out io.Writer, cmd *cobra.Command) error{
+		"bash": runCompletionBash,
+		"zsh":  runCompletionZsh,
+	}
+)
+
+func runCompletion(out io.Writer, cmd *cobra.Command, args []string) error {
+	if len(args) == 0 {
+		return errors.New("shell not specified")
+	}
+	if len(args) > 1 {
+		return errors.New("too many arguments, expected only the shell type")
+	}
+	run, found := completionShells[args[0]]
+	if !found {
+		return errors.Errorf("unsupported shell type %q", args[0])
+	}
+
+	return run(out, cmd)
+}
+
+func runCompletionBash(_ io.Writer, cmd *cobra.Command) error {
+	var buf bytes.Buffer
+	err := cmd.Root().GenBashCompletion(&buf)
+	if err != nil {
+		return fmt.Errorf("error while generating bash completion: %v", err)
+	}
+	// remove the command "completion" from auto-completion
+	code := buf.String()
+	code = strings.Replace(code, `commands+=("completion")`, "", -1)
```

Comment:
Instead of needing to this we can set a `Hidden` property on the completion command object.

(Either globally which would also hide it from the help, or just when actually running the completion command)

Reply:
@ashb it's wrong, how can you run ./astro ---help?

Reply:
```
astro is a command line interface for working with the Astronomer Platform.

Usage:
  astro [command]

Available Commands:
  auth            Manage astronomer identity
  cluster         Manage Astronomer EE clusters
  completion      Generate autocompletions script for the specified shell (bash or zsh)
  config          Manage astro project configurations
```